fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/plans/RecordQueryComparatorPlan.java

```diff
@@ -0,0 +1,233 @@
+        if (children.isEmpty()) {
+            throw new RecordCoreArgumentException("Comparator plan should have at least one plan");
+        }
+        if ((referencePlanIndex < 0) || (referencePlanIndex >= children.size())) {
+            throw new RecordCoreArgumentException("Reference Plan Index should be within the range of sub plans");
+        }
+
+        final ImmutableList.Builder<ExpressionRef<RecordQueryPlan>> childRefsBuilder = ImmutableList.builder();
+        for (RecordQueryPlan child : children) {
+            childRefsBuilder.add(GroupExpressionRef.of(child));
+        }
+        return new RecordQueryComparatorPlan(Quantifiers.fromPlans(childRefsBuilder.build()), comparisonKey, referencePlanIndex);
+    }
+
+    /**
+     * Execute children and compare results.
+     * @param store record store from which to fetch records
+     * @param context evaluation context containing parameter bindings
+     * @param continuation continuation from a previous execution of this same plan
+     * @param executeProperties limits on execution
+     * @param <M> the type of records in the store
+     * @return {@link RecordCursor} that iterates through the results of the execution
+     */
+    @Nonnull
+    @Override
+    @SuppressWarnings("squid:S2095")
+    public <M extends Message> RecordCursor<QueryResult> executePlan(@Nonnull final FDBRecordStoreBase<M> store,
+                                                                     @Nonnull final EvaluationContext context,
+                                                                     @Nullable final byte[] continuation,
+                                                                     @Nonnull final ExecuteProperties executeProperties) {
+        // The child plans all keep their skip and limit - this way we can ensure that they all handle their skip and
+        // limit correctly. The parent plan adds no skip and limit of its own - the reference plan is handling that.
+        final ExecuteProperties parentExecuteProperties = executeProperties.clearSkipAndLimit();
+        return ComparatorCursor.create(store, getComparisonKey(),
+                        getChildren().stream()
+                                .map(childPlan -> comparatorCursorFunction(store, context, executeProperties, childPlan))
+                                .collect(Collectors.toList()),
+                        continuation,
+                        referencePlanIndex,
+                        () -> toString(),
+                        () -> planHash(PlanHashKind.STRUCTURAL_WITHOUT_LITERALS))
+                .skipThenLimit(parentExecuteProperties.getSkip(), parentExecuteProperties.getReturnedRowLimit())
+                .map(QueryResult::of);
+    }
+
+    /*
+     * Return a function that creates a cursor for the given child plan using the provided continuation
+     */
+    @Nonnull
+    private <M extends Message> Function<byte[], RecordCursor<FDBQueriedRecord<M>>> comparatorCursorFunction(
```

Comment:
`comparatorCursorFunction` doesn't seem quite right. I guess `comparandCursorFunction` would work. Or `cursorFunctionForComparator` or something like that.

Or take a different approach like `childCursorFunction`.

Reply:
Done. childCursorFunction seems to capture the meaning best.